### PR TITLE
For linux msm/firehose silence timeout warning

### PR DESCRIPTION
**Pull request by quic-bjorande**

### CODE REVIEW

File: firehose.c [hunk 1]
```diff
@@ -53,6 +53,11 @@
 #include "qdl.h"
 #include "ufs.h"
 
+enum {
+	FIREHOSE_ACK,
```
Comment (konradybcio) [inline firehose.c:57]: "Clean this up by defining ACK and NAK in the positive value space and
use the negative value space for errno."

almost, but I don't mind ;)

maybe add a =value to the first entry

File: firehose.c [hunk 2]
```diff
@@ -106,10 +111,15 @@ static int firehose_generic_parser(xmlNode *node, void *data)
...
-		return 0;
+		return -EAGAIN;
 	}
 
-	return xmlStrcmp(value, (xmlChar*)"ACK") == 0 ? 1 : -1;
+	if (xmlStrcmp(value, (xmlChar*)"ACK") == 0)
+		return FIREHOSE_ACK;
+	if (xmlStrcmp(value, (xmlChar*)"NAK") == 0)
+		return FIREHOSE_NAK;
+
```
Comment (konradybcio) [inline firehose.c:121]: should we print whatever came out here?

Comment (andersson) [reply firehose.c:121]: I think we should follow up and adjust the error prints throughout the file to give better error messages. Perhaps strerror(ret) vs "device rejected" printing.

File: firehose.c [hunk 3]
```diff
@@ -267,14 +277,18 @@ static int firehose_configure(struct qdl_device *qdl, bool skip_storage_init, co
...
+	}
 
 	/* Retry if remote proposed different size */
 	if (size != max_payload_size) {
 		ret = firehose_send_configure(qdl, size, skip_storage_init, storage, &size);
-		if (ret < 0)
-			return ret;
+		if (ret != FIREHOSE_ACK) {
+			fprintf(stderr, "[CONFIGURE] request failed\n");
+			return -1;
```
Comment (konradybcio) [inline firehose.c:286]: might as well return ret to propagate errors from 2 levels deeper

Comment (andersson) [reply firehose.c:286]: No, because then we might return FIREHOSE_NAK from firehose_run(), I rather keep that in {0,-1} space...

**Pull request merged**